Forwarded: part of Origin PR by Jojo-Schmitz
Origin: https://github.com/musescore/MuseScore/pull/9000
Author: Gabriel Vogel <gabriel.vogel@online.de>
Description: Fix #324840: Ctrl+Scrollwheel zoom on Linux X11
 With recent libinput versions zooming with Ctrl+Scrollwheel would zoom
 in huge steps. This change enforces the previous behavior for Linux
 only.

--- a/mscore/events.cpp
+++ b/mscore/events.cpp
@@ -110,6 +110,15 @@ void ScoreView::wheelEvent(QWheelEvent*
       int dx = 0, dy = 0, n = 0;
       qreal nReal = 0.0;
 
+// pixelDelta is unreliable on X11
+#ifdef Q_OS_LINUX
+      if (std::getenv("WAYLAND_DISPLAY") == NULL) {
+          // Ignore pixelsScrolled unless Wayland is used
+          pixelsScrolled.setX(0);
+          pixelsScrolled.setY(0);
+      }
+#endif
+
       if (!pixelsScrolled.isNull()) {
             dx = pixelsScrolled.x();
             dy = pixelsScrolled.y();
